types/denom.go

```diff
@@ -20,6 +23,10 @@ func RegisterDenom(denom string, unit Dec) error {
 	}
 
 	denomUnits[denom] = unit
+
+	if baseDenom == nil || unit.LT(denomUnits[*baseDenom]) {
```

Comment:
We can avoid using a pointer if the default is just `""` right?

Reply:
I'm not sure which one is more idiomatic in golang, not very experience in golang yet.

Reply:
done